### ISSUE TITLE
errors with RadioGroup component

### ISSUE THREAD

**Issue by JameelKhan9** [tailwindlabs/headlessui#2321]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

v1.7.12

**What browser are you using?**

N/A

**Reproduction URL**
The current react playground can be updated to have extra props from `RadioPropsWeControl` added to reproduce the issue.

**Describe your issue**

Upgrading from v1.7.11 to 1.7.12 caused some type errors with RadioGroup component.

![image](https://user-images.githubusercontent.com/59159249/222067034-46320cd8-d37e-4c06-b059-2d7eb5d0ad36.png)

I believe the error might be caused by the changes made here:
https://github.com/tailwindlabs/headlessui/pull/2282/files#diff-1197905c3c8166d41d6fd6273126ee1d53f55086adff65c49c120495e1bc8634L383

**Comment by JameelKhan9** [comment]
@thecrypticace let me know your thoughts. I am happy to contribute with some guidance.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

We recently made some changes to the TypeScript types so that they better reflect reality. In this particular case, the `tabIndex` is already handled by us inside Headless UI so adding a dedicated `tabIndex` prop doesn't do anything or shouldn't change the behaviour.

In fact, if you change the `tabIndex` to `123` in `1.7.11` you shouldn't see them in the DOM ever so while the types are more correct / explicit, the behaviour should not change.

In other words, you should be able to drop that prop altogether and it should work as is.

Is there a specific reason you added those `tabIndex` props yourself?

If you _do_ need different behaviour can you open an issue to talk about the behaviour you want and also provide a minimal reproduction repo that showcases the issue?

Hope this helps!

**Comment by JameelKhan9** [comment]
Hi @RobinMalfait thanks for the well-formed explanation! Makes sense.

I was just upgrading modules and found the error in a component someone else made. On a closer look, it doesn't look like there is much of a use case around specifying this index. I will go ahead and remove it and will open an issue if the behavior is required in the future for any reason.

Thanks again.
